pkg/instancetype/instancetype.go

```diff
@@ -852,6 +854,18 @@ func (m *InstancetypeMethods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1
 	return "", "", fmt.Errorf("unable to infer defaults from DataVolumeSourceRef as Kind %s is not supported", sourceRef.Kind)
 }
 
+func applyAnnotations(annotations map[string]string, target metav1.Object) Conflicts {
```

Comment:
We will need two functions to handle the different use cases IMHO.

Annotations from an instance type should conflict if an existing annotation with a different value exists.

Annotations from a preference should not conflict if an existing annotation exists and should not be applied.

Reply:
+1, I'd also like to see unit tests covering these cases when implemented.